### PR TITLE
E_CLASSROOM-286 (Dependent #288) [FE/BE] Categories sort by column

### PR DESCRIPTION
**Pull request by ErickDoroy**
### Issue Link
- https://framgiaph.backlog.com/view/E_CLASSROOM-286
### Definition of Done
- [x] User should be able to sort by id, name
- [x] Refreshing the page should keep the sort option
- [x] First click should sort by ascending, second click on same column, sort by descending and third click is default
- [x] Arrow icon should be visible when the column is sorted

## Related PR:
- https://github.com/framgia/sph-classroom-els-be/pull/67
#### Main note
- http://localhost:3003/admin/quizzes?page=1&sortBy=&sortDirection=
#### Pages Affected
- Category Page

### Scenarios/ Test Cases
- [x] it should sort the table by name or id
#### User Interface
- N/A
#### User Experience 
- N/A
#### Functionality
- N/A
### Screenshots
categoryadmin

### CODE REVIEW

File: src/pages/admin/categories/CategoryList/index.js [hunk 1]
```diff
@@ -16,67 +16,78 @@ import Container from 'react-bootstrap/Container';
...
   const queryParams = new URLSearchParams(window.location.search);
   const pageNum = queryParams.get('page');
+  const sortBy = queryParams.get('sortBy') || '';
+  const sortDirection = queryParams.get('sortDirection') || '';
 
   //TEMPORARY VALUES since the search, sort, and filter functionality isn't implemented yet.
-  const sortBy = 'asc';
-  const filter = '';
+  // const sortBy = 'asc';
+  // const filter = '';
```
Comment (karlolee50) [inline src/pages/admin/categories/CategoryList/index.js]: remove if not used

File: src/api/Category/index.js [hunk 2]
```diff
@@ -74,6 +74,18 @@ const CategoryApi = {
     return API.request(options);
   },
 
+  listOfCategories: ({ ...params }) => {
+    const options = {
+      method: 'GET',
+      url: '/admin/categories-data',
```
Comment (karlolee50) [inline src/api/Category/index.js]: don't forget to update this URL as well after updating BE

**Review by karlolee50 — CHANGES_REQUESTED** [review]
Review time: 10 minutes
Comment/s: 2

**Comment by Teraaay** [comment]
### Working
- [x] Able to sort by ID and Name
- [x] Refreshing the page keeps the sort options
- [x] First click sorts the list in ascending order, second click sorts the list in descending order, third click sorts the list back to its original order.
- [x] Corresponding arrow icons are visible when clicking a column name

### Comment/s
- [x] Please revert the UI back to its original design.
- [x] Please remove the sort option of the Description Column since a user should only be able to sort by ID and Name.
- [x] There should be a spinner and loading message every time a user sorts the list, to indicate that the data is still being loaded.

_refer to the screenshots below_

### Screenshot/s
SORT CATEGORIES BY COL COMMENT

image

There should be a spinner and loading message every time a user sorts the list, example screenshot below:
> image

------
- Total Test Time: 0.25 hr
- Total Comment/s: 0
- Total Iterations: 0

**Comment by Teraaay** [comment]
### Approved PR #131 
- [x] Able to sort by ID and Name
- [x] Refreshing the page keeps the sort options
- [x] First click sorts the list in ascending order, second click sorts the list in descending order, third click sorts the list back to its original order.
- [x] Corresponding arrow icons are visible when clicking a column name

Able to fix the comments:
 
- [x] Able to revert back to its original design.
- [x] User cannot sort by description.
- [x] There is a spinner and a loading message, every time a user sorts.

### Screenshot/s
> SORT CATEGORIES BY COL FINALgif

----
- Total Test time: 0.5 hr
- Total Iterations: 1
- Total Comments: 3

**Review by karlolee50 — APPROVED** [review]

**Pull request merged**